Debugger crashing on stop

There's a crash on stopping emulator due to listener being nil.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x28 pc=0x10327f3f4]

goroutine 177 [running]:
github.com/onflow/flow-emulator/server/debugger.(*Debugger).Stop.func1()
	/Users/dapper/go/pkg/mod/github.com/onflow/flow-emulator@v0.44.1/server/debugger/debugger.go:120 +0x34
sync.(*Once).doSlow(0x14000be8690?, 0x103273620?)
	/usr/local/go/src/sync/once.go:74 +0x104
sync.(*Once).Do(...)
	/usr/local/go/src/sync/once.go:65
github.com/onflow/flow-emulator/server/debugger.(*Debugger).Stop(0x103ceafe0?)
	/Users/dapper/go/pkg/mod/github.com/onflow/flow-emulator@v0.44.1/server/debugger/debugger.go:118 +0x48
github.com/psiemens/graceland.(*Group).Stop(0x140014fd900)
	/Users/dapper/go/pkg/mod/github.com/psiemens/graceland@v1.0.0/graceland.go:112 +0x84
github.com/psiemens/graceland.(*Group).Start.func1(0x0?, {0x103fb5240?, 0x140014fd600?})
	/Users/dapper/go/pkg/mod/github.com/psiemens/graceland@v1.0.0/graceland.go:80 +0x68
created by github.com/psiemens/graceland.(*Group).Start
	/Users/dapper/go/pkg/mod/github.com/psiemens/graceland@v1.0.0/graceland.go:70 +0x27c
```

--- a/server/debugger/debugger.go
+++ b/server/debugger/debugger.go
@@ -117,6 +117,8 @@ func (d *Debugger) handleConnection(conn net.Conn) {
 func (d *Debugger) Stop() {
 	d.stopOnce.Do(func() {
 		close(d.quit)
-		d.listener.Close()
+		if d.listener != nil {
+			d.listener.Close()
+		}
 	})
 }

